pyls/python_ls.py

```diff
@@ -39,8 +40,10 @@ def capabilities(self):
             'textDocumentSync': lsp.TextDocumentSyncKind.INCREMENTAL
         }
 
-    def initialize(self, root_path, init_opts, _process_id):
-        self.workspace = Workspace(root_path, lang_server=self)
+    def initialize(self, root_uri, init_opts, _process_id):
+        self.workspace = Workspace(root_uri, lang_server=self)
+
+        root_path = unquote(urlparse(root_uri).path)
         self.config = config.Config(root_path, init_opts)
```

Comment:
Yeah, we kind of lost support for non-local workspaces in the plug upgrade. happy to add it back one day. 

Since we're pre-1.0 I wouldn't worry too much about API breaks. So I'd just replace the Config field with root_uri (not path), have Workspace also take a root_uri, and then parse the path once in the initialisation opts